sqlglot/dialects/bigquery.py

```diff
@@ -645,14 +645,18 @@ def _parse_table_part(self, schema: bool = False) -> t.Optional[exp.Expression]:
 
                     table_name += self._find_sql(start, self._prev)
 
+                previous: exp.Expression = this
                 this = exp.Identifier(this=table_name, quoted=this.args.get("quoted"))
+                this.update_positions(previous)
```

Comment:
I think you can also do something like `exp.Identifier(...).update_positions(this)` if you want to.